dali/operators/image/distortion/jpeg_compression_distortion_op_cpu.cc

```diff
@@ -68,17 +100,22 @@ void JpegCompressionDistortionCPU::RunImpl(workspace_t<CPUBackend> &ws) {
   thread_ctx_.resize(thread_pool.NumThreads());
 
   for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
-    thread_pool.AddWork(
-      [&, sample_idx, quality = quality_arg_[sample_idx].data[0]](int thread_id) {
-        auto &ctx = thread_ctx_[thread_id];
-        auto sh = in_shape.tensor_shape_span(sample_idx);
-        cv::Mat in_mat(sh[0], sh[1], CV_8UC3, const_cast<unsigned char*>(in_view[sample_idx].data));
-        cv::Mat out_mat(sh[0], sh[1], CV_8UC3, out_view[sample_idx].data);
-        cv::cvtColor(in_mat, out_mat, cv::COLOR_RGB2BGR);
-        cv::imencode(".jpg", out_mat, ctx.encoded, {cv::IMWRITE_JPEG_QUALITY, quality});
-        cv::imdecode(ctx.encoded, cv::IMREAD_COLOR, &out_mat);
-        cv::cvtColor(out_mat, out_mat, cv::COLOR_BGR2RGB);
-      }, in_shape.tensor_size(sample_idx));
+    auto shape = layoutShape(input.GetLayout(), in_shape.tensor_shape_span(sample_idx));
+    for (int elem_idx = 0; elem_idx < shape[0]; elem_idx++) {
+      int height = shape[1];
+      int width = shape[2];
+      int channels = shape[3];
+      DALI_ENFORCE(channels == 3 && "Supports 3 channel images");
```

Comment:
We plan to remove the assertion condition from the error message sooner or latter so I would make sure the message itself tells where the problem is.
```suggestion
      DALI_ENFORCE(channels == 3, "Only 3 channel images are supported");
```

Reply:
Done